### ISSUE TITLE
[fmlrc-convert] ERROR - unexpected symbol in input: char: " ", hex: "20"

### ISSUE THREAD

**Issue by pezhmansafdari** [holtjma/fmlrc#15]
Hi,

when I run the 
gunzip -c reads.sorted.txt.gz | tr NT TN | /opt/ropebwt2/ropebwt2 -LR | tr NT TN | /opt/fmlrc-1.0.0/fmlrc-convert comp_msbwt.npy
command, I get the following error:
ropebwt2: mrope.c:268: mr_insert_multi: Assertion `len > 0 && s[len-1] == 0' failed.
[fmlrc-convert] ERROR - unexpected symbol in input: char: " ", hex: "20"
Any idea what the source of the error might be?
BR,
Pezhman

**Comment by holtjma** [comment]
Generally speaking, errors like this are caused by unexpected/malformed read inputs.  For example, to replicate the error generated above, create a read file and append a single space (" ") to the end of one of the reads like so (here the second sequence is "G "):
```
A
G 
CT
```
If you run the following command using the above "bad_reads.txt" file:
```
cat bad_reads.txt | sort | tr NT TN | ./example/ropebwt2/ropebwt2 -LR | tr NT TN | ./fmlrc-convert -f ./comp_msbwt.npy
```
then the resulting output replicates the errors you have above:
```
Assertion failed: (len > 0 && s[len-1] == 0), function mr_insert_multi, file mrope.c, line 268.
[fmlrc-convert] Reading from stdin
[fmlrc-convert] ERROR - unexpected symbol in input: char: " ", hex: "20"
```

In summary, I would check for leading/trailing characters in your read file and remove them.

**Comment by holtjma** [comment]
Closing due to inactivity.
